eq-author-api/schema/tests/calculatedSummary.test.js

```diff
@@ -0,0 +1,548 @@
+      id: questionnaire.sections[0].pages[1].id,
+      summaryAnswers: [
+        questionnaire.sections[0].pages[0].answers[0].id,
+        questionnaire.sections[0].pages[0].answers[1].id,
+      ],
+    });
+
+    await updateCalculatedSummaryPage(questionnaire, {
+      id: questionnaire.sections[0].pages[1].id,
+      summaryAnswers: [questionnaire.sections[0].pages[0].answers[0].id],
+    });
+
+    const result = await queryPage(questionnaire, page.id);
+
+    expect(result).toMatchObject({
+      id: expect.any(String),
+      summaryAnswers: [
+        { id: questionnaire.sections[0].pages[0].answers[0].id },
+      ],
+    });
+  });
+
+  it("should delete a calculated summary", async () => {
+    const questionnaire = await buildQuestionnaire({
+      sections: [{ pages: [{ pageType: "calculatedSummary" }] }],
+    });
+
+    const section = questionnaire.sections[0];
+    const page = section.pages[0];
+
+    await deletePage(questionnaire, page.id);
+
+    const result = await querySection(questionnaire, section.id);
+
+    expect(result.pages).toHaveLength(0);
+  });
+
+  it("should return a list of avalible summary answers", async () => {
+    const questionnaire = await buildQuestionnaire({
+      sections: [
+        {
+          pages: [
+            {
+              answers: [
+                {
+                  type: NUMBER,
+                },
+                {
+                  type: NUMBER,
+                },
+                {
+                  type: NUMBER,
+                },
+                {
+                  type: CURRENCY,
+                },
+              ],
+            },
+            {
+              pageType: "calculatedSummary",
+            },
+            {
+              answers: [
+                {
+                  type: NUMBER,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    });
+    const answersPage = questionnaire.sections[0].pages[0];
+    const calSumPage = questionnaire.sections[0].pages[1];
+
+    await updateCalculatedSummaryPage(questionnaire, {
+      id: calSumPage.id,
+      summaryAnswers: [answersPage.answers[0].id],
+    });
+
+    const result = await queryPage(questionnaire, calSumPage.id);
+
+    expect(result).toMatchObject({
+      id: expect.any(String),
+      availableSummaryAnswers: [
+        { id: answersPage.answers[1].id },
+        { id: answersPage.answers[2].id },
+      ],
+    });
+  });
+
+  it("should return more than one type of summary answeres if more than one type selected", async () => {
+    const questionnaire = await buildQuestionnaire({
+      sections: [
+        {
+          pages: [
+            {
+              answers: [
+                {
+                  type: NUMBER,
+                },
+                {
+                  type: CURRENCY,
+                },
+              ],
+            },
+            {
+              pageType: "calculatedSummary",
+            },
+            {
```

Comment:
Is this page ever used?

Reply:
That last page is there to ensure that pages and their answers after the calSum aren't present in the availableAnswers. I've updated the test to make it more explicit and explain the test a bit more.